aimmo-game/simulation/worker_manager.py

```diff
@@ -30,22 +44,64 @@ def run(self):
                         # for turns
                         self.game_state.remove_avatar(user['id'])
                         # Get persistent state from worker
-                        # TODO
+                        persistent_state = self.get_persistent_state(user['id'])
                         # Kill worker
-                        # TODO
+                        self.remove_worker(user['id'])
                         # Spawn worker
-                        # Comes from spawning
-                        worker_url = 'http://localhost:%d' % (
-                            5000 + int(user['id']))
-                        # TODO
+                        worker_url = self.create_worker(
+                            user['id'],
+                        )
                         # Initialise worker
                         requests.post("%s/initialise/" % worker_url, json={
                             'code': user['code'],
                             'options': {},
+                            'persistent_state': persistent_state,
                         })
                         # Add avatar back into game
                         self.game_state.add_avatar(
                             user_id=user['id'], worker_url="%s/turn/" % worker_url)
                         self.user_codes[user['id']] = user['code']
+                removed_user_ids = set(self.user_codes) - set(user['id'] for user in game['users'])
+                for removed_user in removed_user_ids:
+                    self.game_state.remove_avatar(removed_user)
+                    self.remove_worker(removed_user)
+                    del self.user_codes[removed_user]
 
             time.sleep(10)
+
+class LocalWorkerManager(WorkerManager):
+    '''Relies on them already being created already.'''
+    host = '127.0.13.4'
+    worker_path = os.path.join(
+        os.path.dirname(__file__),
+        '../../aimmo-game-worker/service.py' ,
+    )
+
+    def __init__(self, *args, **kwargs):
+        self.workers = {}
+        self.next_port = 1989
+        super(LocalWorkerManager, self).__init__(*args, **kwargs)
+
+    def create_worker(self, player_id):
+        if player_id in self.workers:
+            raise NotImplemented
+        self.next_port += 1
+        self.workers[player_id] = subprocess.Popen(
+            [
+                self.worker_path,
+                self.host,
+                str(self.next_port),
+            ],
+        )
+        worker_url = 'http://%s:%d' % (
+            self.host,
+            self.next_port,
+        )
+        LOGGER.info("Worker started for %s, listening at %s", player_id, worker_url)
+        time.sleep(5)
```

Comment:
Aye. This should be some sort of retry logic, and initialisation should probably be split off from worker creation. I'll raise another issue for this